changelog.d/16506.bugfix

```diff
@@ -0,0 +1 @@
+Fix a bug introduced in Synapse 1.59.0 where servers would be incorrectly marked as available when a request resulted in an error.
```

Comment:
I'm not sure if there's a more visible symptom? Perhaps it would cause things to be retried too often?